src/manifolds/EmbeddedManifold.jl

```diff
@@ -0,0 +1,450 @@
+"""
+    AbstractEmbeddingType
+
+A type to provide properties of the embedding an [`AbstractEmbeddedManifold`](@ref) has.
+"""
+abstract type AbstractEmbeddingType end
+
+"""
+    AbstractEmbeddedManifold{T<:AbstractEmbeddingType} <: AbstractDecoratorManifold
+
+An abstract type for an EmbeddedManifold, which acts as an [`AbstractDecoratorManifold`](@ref).
+The functions of the manifold that is embedded can hence be just passed on to the embedding.
+The embedding is further specified by an [`AbstractEmbeddingType`](@ref).
+
+This means, that formally an embedded manifold is a decorator for the manifold it is embedded
+into.
+"""
+abstract type AbstractEmbeddedManifold{T<:AbstractEmbeddingType} <: AbstractDecoratorManifold end
+
+"""
+    AbstractIsometricEmbeddingType <: AbstractEmbeddingType
+
+Characterizes an that an embedding is isometric. For this case the [`inner`](@ref) product
+is passed from the embedded manifold to the embedding.
+"""
+abstract type AbstractIsometricEmbeddingType <: AbstractEmbeddingType end
+
+"""
+    DefaultIsometricEmbedding <: AbstractIsometricEmbeddingType
+
+Specify that an embedding is the default isometric embedding. This even inherits
+logarithmic and exponential map as well as retraction and inverse retractions from the
+embedding.
```

Comment:
"From the embedding" or from the decorated manifold or something else?

Reply:
When I was thinking about default embeddings I had something like the default metric in mind and this looks like a manifold that is a linear subspace of its embedding? I'm a bit confused now.

Reply:
Note that this is the `DefaultIsometricEmbedding` i.e. even a special __isometric__ embedding. We can name that differently, but we need a type like this to distinguish (already mentioned above) symmetric, skew-symmetric and such (inheriting even exp/log) from those that don't (like the sphere).

Reply:
It actually inherits exp and log from the embedding – if the embedding is a manifold with a non-default-metric – that's fine.

Reply:
OK, so we probably need two different types here, one that acts like the default metric and another one for inheriting `exp` and other related functions. For the second one I'd prefer a different name than `DefaultIsometricEmbedding`, maybe `TransparentIsometricEmbedding`?

Reply:
I took default since it's “just” a little more transparent then the one without. But of course it's a slight misuse of the default terminology to the other cases. I looked for a name but could not find a good name. We can take `TransparentIsometricEmbedding` for sure and document that that's meant for those where eps/log... are even transparently passed on.